### DISCUSSION TITLE
How can I rid of all JavaScript?

### DISCUSSION THREAD

**Discussion by ghost** [squidfunk/mkdocs-material#6593] [Q&A] [answered]
Hi, I'm loving Material, but my site does not need JS nor Search functionality.
My readers are supposed to be Mass-Surveillance aware people (hopefully) and they are not going to run any JS.
(It will actually look bad that I try to run JS in someone else browsers)

Also LibreJS will block anyway several JS from loading reporting some warnings.

Thank you in advance.
Chip

![Screenshot from 2024-01-04 11-34-00](https://github.com/squidfunk/mkdocs-material/assets/119405768/48cb2673-fa08-4624-a3a8-15b67c70a755)

**Comment by kamilkrzyskow** [comment, marked as answer]
Hello @alithechemist,
you can override the theme and prevent adding JavaScript to the HTML code:
https://squidfunk.github.io/mkdocs-material/customization/#extending-the-theme
https://squidfunk.github.io/mkdocs-material/customization/#overriding-blocks
https://github.com/squidfunk/mkdocs-material/blob/246266fafad2c2fdc78022c14f7f4b2df91167f3/src/templates/base.html#L389-L441

Override the `config` and `scripts` blocks ans keep them empty.

If you require that no `bundle.js` is even included in the deployed files, then you would have to add a hook that removes the file from the rendered `site` directory:
https://www.mkdocs.org/user-guide/configuration/#hooks

If you need more help I could stitch together some example.

**Reply by kamilkrzyskow** [reply]
Oh also looking at your error, on the screenshot I see line 41 non-trivial localStorage. 
To remove that you also have to override the `partials/javascripts` files and keep them empty https://github.com/squidfunk/mkdocs-material/tree/master/src/templates/partials/javascripts, as I believe not every usage instance is in a convenient `block` you could override in `main.html`. `partials/javascripts/base.html` should be enough, as other refer to special features, but I guess you could just be extra careful to prevent any issues with blocked JavaScript.

**Reply by ghost** [reply]
Hi, thanks, for the super quick reply.
I'm actually trying to override the blocks as you suggested, but I end up with a white page.

What I did.
 - create a folder 'overrides' in the same folder that contains mkdocs.yml
 - add a file called base.html
 - add content from mkdocs-material/src/templates/base.html
 - remove the content between the two blocks like you suggested.

content of overrides/base.html
```
{% block config %} 
{% endblock %} 
  
 <!-- Theme-related JavaScript --> 
 {% block scripts %} 
 {% endblock %}
```


Am I missing something?
Thanks in advance.

**Reply by ghost** [reply]
oopsy, Just got it working, I had to call the override "main.html" and use 
```
{% extends "base.html" %}
{% block config %} 
{% endblock %} 
  
 <!-- Theme-related JavaScript --> 
 {% block scripts %} 
 {% endblock %}
```
Everything working as expected.

Also there is one Javascript used by mkdocs to reload the pages when I change the source, that's ok.

Thanks a lot!
Chip

**Reply by ghost** [reply]
@kamilkrzyskow me again,
I have now the following directory tree:
```
mkdocs.yml
overrides
├── main.html
└── partials
    └── javascript
        ├── announce.html
        ├── base.html
        ├── consent.html
        ├── content.html
        ├── outdated.html
        └── palette.html
```
all the files in partials/javascripts/* are empty.

**But still I see the script in the LibreJS warning (the one on line 41 in the image I attached)
I did not see it before...**

As you suggested the JS is  in base.html and I created an empty file in overrides/javascripts. 
Are these directories correct for partials overrides?
What might be the reason why this one is not working?
I did not see any cache...

Thanks in advance for any suggestions.
Chip

**Reply by kamilkrzyskow** [reply]
Based on the message I see that you created a `partials/javascript` directory instead of `partials/javascripts`

> As you suggested the JS is in base.html and I created an empty file in overrides/javascripts.

`overrides/javascripts` also isn't correct. 

You have to map the directory tree 1:1 to override it.

https://github.com/squidfunk/mkdocs-material/blob/246266fafad2c2fdc78022c14f7f4b2df91167f3/src/templates/base.html#L152-L155

**Reply by ghost** [reply]
Renaming javascript to javascripts made it work.
Thanks again, sorry for the rookie mistake.
Chip
